src/navigators/LoggedInStack.tsx

```diff
@@ -0,0 +1,129 @@
+import { OnboardingCourseScreen } from '../screens/OnboardingCourseScreen';
+import { createNativeStackNavigator } from '@react-navigation/native-stack';
+import { LoggedInRootParamList } from './types';
+import { useConsent } from '../hooks/useConsent';
+import { useActiveAccount } from '../hooks/useActiveAccount';
+import { useActiveProject } from '../hooks/useActiveProject';
+import { useOnboardingCourse } from '../hooks/useOnboardingCourse';
+import { ActivityIndicatorView } from '../components/ActivityIndicatorView';
+import { t } from 'i18next';
+import { usePendingInvite } from '../hooks/usePendingInvite';
+import { useSetUserProfileEffect } from '../hooks/useSetUserProfileEffect';
+
+export function LoggedInStack() {
+  const Stack = createNativeStackNavigator<LoggedInRootParamList>();
+  const { inviteParams } = usePendingInvite();
+  useSetUserProfileEffect();
+  const {
+    account,
+    isLoading: isLoadingAccount,
+    isFetched: isFetchedAccount,
+  } = useActiveAccount();
+  const {
+    activeProject,
+    isLoading: isLoadingProject,
+    isFetched: isFetchedProject,
+  } = useActiveProject();
+  const { useShouldRenderConsentScreen } = useConsent();
+  const { shouldRenderConsentScreen, isLoading: loadingConsents } =
+    useShouldRenderConsentScreen();
+  const {
+    shouldLaunchOnboardingCourse,
+    isLoading: onboardingCourseIsLoading,
+    isFetched: onboardingCourseIsFetched,
+  } = useOnboardingCourse();
+
+  const loadingProject = !isFetchedProject || isLoadingProject;
+  const loadingAccount = !isFetchedAccount || isLoadingAccount;
+  const hasAccount = !loadingAccount && !!account?.id;
+  const loadingAccountOrProject =
+    loadingAccount || (hasAccount && loadingProject);
+  const loadingOnboardingCourse =
+    !onboardingCourseIsFetched || onboardingCourseIsLoading;
+
+  const hasAccountAndProject = !!(activeProject?.id && account?.id);
+  const initialRoute = !hasAccountAndProject
+    ? 'InviteRequired'
+    : shouldRenderConsentScreen
+    ? 'screens/ConsentScreen'
+    : shouldLaunchOnboardingCourse
+    ? 'screens/OnboardingCourseScreen'
+    : 'app';
+
+  if (loadingAccountOrProject) {
+    return (
+      <ActivityIndicatorView
+        message={t(
+          'root-stack-waiting-for-account-and-project',
+          'Loading account',
+        )}
+      />
+    );
+  }
+
+  if (hasAccountAndProject && loadingConsents) {
+    return (
+      <ActivityIndicatorView
+        message={t('root-stack-waiting-for-consents', 'Loading consents')}
+      />
+    );
+  }
+
+  if (hasAccountAndProject && loadingOnboardingCourse) {
+    return (
+      <ActivityIndicatorView
+        message={t(
+          'root-stack-waiting-for-app-config',
+          'Loading onboarding course data',
+        )}
+      />
+    );
+  }
+
+  if (inviteParams?.inviteId) {
+    return (
+      <ActivityIndicatorView
+        message={t('root-stack-accepting-invitation', 'Accepting invitation')}
+      />
+    );
+  }
```

Comment:
Maybe not in this PR but I think it would be good to just have one `ActivityIndicatorView` and change the message depending on the state. By returning different components we will see the indicator restart instead of just spinning smoothly.

Reply:
That's an excellent suggestion and would clean up the code too. I'll grab that in a future PR.